backends/xboxone/app-template/App.xaml.cs

```diff
@@ -0,0 +1,104 @@
+ */
+
+// This is the C# code-behind for App.xaml.  It's all boilerplate generated by
+// Visual Studio, then cleaned up by hand to remove things we definitely didn't
+// need.  There may be more cleanup possible.
+
+using System;
+using Windows.ApplicationModel;
+using Windows.ApplicationModel.Activation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Navigation;
+
+namespace XboxOneWebDriverServer
+{
+    /// <summary>
+    /// Provides application-specific behavior to supplement the default Application class.
+    /// </summary>
+    sealed partial class App : Application
+    {
+        /// <summary>
+        /// Initializes the singleton application object.  This is the first line of authored code
+        /// executed, and as such is the logical equivalent of main() or WinMain().
+        /// </summary>
+        public App()
+        {
+            this.InitializeComponent();
+            this.Suspending += OnSuspending;
+        }
+
+        /// <summary>
+        /// Invoked when the application is launched normally by the end user.  Other entry points
+        /// will be used such as when the application is launched to open a specific file.
+        /// </summary>
+        /// <param name="e">Details about the launch request and process.</param>
+        protected override void OnLaunched(LaunchActivatedEventArgs e)
+        {
+            Frame rootFrame = Window.Current.Content as Frame;
+
+            // Do not repeat app initialization when the Window already has content,
+            // just ensure that the window is active
+            if (rootFrame == null)
+            {
+                // Create a Frame to act as the navigation context and navigate to the first page
+                rootFrame = new Frame();
+
+                rootFrame.NavigationFailed += OnNavigationFailed;
+
+                // Place the frame in the current Window
+                Window.Current.Content = rootFrame;
+            }
+
+            if (e.PrelaunchActivated == false)
+            {
+                if (rootFrame.Content == null)
+                {
+                    // When the navigation stack isn't restored navigate to the first page,
+                    // configuring the new page by passing required information as a navigation
+                    // parameter
+                    rootFrame.Navigate(typeof(MainPage), e.Arguments);
+                }
+                // Ensure the current window is active
+                Window.Current.Activate();
+            }
+        }
+
+        /// <summary>
+        /// Invoked when Navigation to a certain page fails
+        /// </summary>
+        /// <param name="sender">The Frame which failed navigation</param>
+        /// <param name="e">Details about the navigation failure</param>
+        void OnNavigationFailed(object sender, NavigationFailedEventArgs e)
+        {
+            throw new Exception("Failed to load Page " + e.SourcePageType.FullName);
```

Comment:
What are you trying to do here?  In a normal app, when an uncaught exception is thrown, the app will close.  I don't know if this would show anything to the user except just close the app.

Reply:
This is part of the MS-generated boilerplate.  I didn't write this, and I don't know what it does.  I don't even know how/when this event could happen.  I tried to explain in the comment at the top of the file:

```csharp
// This is the C# code-behind for App.xaml.  It's all boilerplate generated by
// Visual Studio, then cleaned up by hand to remove things we definitely didn't
// need.  There may be more cleanup possible.
```

I can experiment with removing more of this to see how much of it we actually need.

Reply:
I was able to remove all of that, and it still works.  Thanks for the suggestion!